core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -209,11 +212,43 @@ oppia.factory('TopicEditorStateService', [
        * _subtopicPages list.
        */
       setSubtopicPage: function(subtopicPage) {
-        if (_getSubtopicPageIndex(subtopicPage.getId()) !== -1) {
+        if (_getSubtopicPageIndex(subtopicPage.getId()) !== null) {
           _subtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
             angular.copy(subtopicPage);
         } else {
           _setSubtopicPage(subtopicPage);
+          _newSubtopicPages.push(angular.copy(subtopicPage));
+        }
+      },
+
+      deleteSubtopicPage: function(topicId, subtopicId) {
+        var subtopicPageId = topicId + '-' + subtopicId.toString();
+        var index = _getSubtopicPageIndex(subtopicPageId);
+        var newIndex = _getNewSubtopicPageIndex(subtopicPageId);
+        if (index !== null) {
```

Comment:
What if index is null (and why would it be)? Shouldn't that cause a thrown error due to invalidity?

Reply:
Done.